judge/signals.py

```diff
@@ -120,13 +120,29 @@ def organization_update(sender, instance, **kwargs):
 _misc_config_i18n.append('')
 
 
-@receiver(post_save, sender=MiscConfig)
-def misc_config_update(sender, instance, **kwargs):
-    cache.delete_many(['misc_config:%s:%s:%s' % (domain, lang, instance.key.split('.')[0])
+def misc_config_cache_delete(key):
+    cache.delete_many(['misc_config:%s:%s:%s' % (domain, lang, key.split('.')[0])
                        for lang in _misc_config_i18n
                        for domain in Site.objects.values_list('domain', flat=True)])
 
 
+@receiver(pre_save, sender=MiscConfig)
+def misc_config_update(sender, instance, **kwargs):
+    misc_config_cache_delete(instance.key)
+    try:
+        old_key = MiscConfig.objects.get(id=instance.id).key
+    except MiscConfig.DoesNotExist:
+        pass
+    else:
+        if old_key != instance.key:
+            misc_config_cache_delete(old_key)
```

Comment:
Isn't this a race? Since this is pre-save, a query could happen just before it saves and cache the old value again.

Reply:
What would be a solution to fix this?

One solution might be to grab the old key in `pre_save` and clear the cache in `post_save`.

Reply:
The race should be fixed now by grabbing the old key in `pre_save` and clearing the cache in `post_save`.